skbio/metadata/_interval.py

```diff
@@ -828,6 +828,9 @@ def query(self, bounds=None, metadata=None):
         Interval
             ``Interval`` object satisfying the search criteria.
         """
+        if bounds is None and metadata is None:
+            metadata = {}
```

Comment:
Thanks -- I've tested this locally, and confirm that this works (now `i.query() == i.query(metadata={})`. However, I believe that the code can be simplified by removing the `bounds is None` check on this line:

```python
if metadata is None:
    metadata = {}
```

This will also let us delete the identical check of `if metadata is None` on lines 840 and 841. Since that check is run through every iteration of the `for loc in bounds` loop, this may add a tiny performance boost (if nothing else, it'll make the code easier to read).

Reply:
Thanks! Deferring this as at the time the focus is correctness of function